File: backend/src/order/objects/OrderPriceBreakdown.scala
package delivery.order.objects

final case class OrderPriceBreakdownLine(
    key: String,
    label: String,
    amount: Double,
    kind: String
)

final case class OrderPriceBreakdown(
    lines: List[OrderPriceBreakdownLine],
    productOriginalAmount: Double,
    merchantDiscountAmount: Double,
    voucherDiscountAmount: Double,
    platformDiscountAmount: Double,
    deliveryFeeAmount: Double,
    discountAmount: Double,
    payableAmount: Double
)

File: frontend/src/objects/order/OrderPriceBreakdown.ts
export type OrderPriceBreakdownLineKind = 'charge' | 'discount' | 'total'

export interface OrderPriceBreakdownLine {
  key: string
  label: string
  amount: number
  kind: OrderPriceBreakdownLineKind
}

export interface OrderPriceBreakdown {
  lines: OrderPriceBreakdownLine[]
  productOriginalAmount: number
  merchantDiscountAmount: number
  voucherDiscountAmount: number
  platformDiscountAmount: number
  deliveryFeeAmount: number
  discountAmount: number
  payableAmount: number
}
